src/devices/DCMotor/DCMotor2PinNoEnable.cs

```diff
@@ -77,9 +77,9 @@ public override double Speed
         }
         public override void Dispose()
         {
-            base.Dispose();
             _pwm?.Dispose();
             _pwm = null;
+            base.Dispose();
```

Comment:
This is wrong as it is not implementing the dispose pattern. We should instead have a public Dispose() only in the base class, which calls a protected virtual void Dispose(bool disposing) and that should be the one that is overriden in subclasses.